### ISSUE TITLE
DataChart - detail is not accessible via keyboard when using voiceover

### ISSUE THREAD

**Issue by jcfilben** [grommet/grommet#7319]
### Expected Behavior
When using voice over you should be able to access the details layer via keyboard.

### Actual Behavior
Outside of voice over DataChart details can be access with the left and right arrow keys. However with voiceover running there is not a way to access the details with the keyboard.

### URL, screen shot, or Codepen exhibiting the issue
This issue can be seen in this story while using voiceover
https://storybook.grommet.io/?path=/story/visualizations-datachart-detail--detail

### Steps to Reproduce

1. 2. 3.

### Your Environment

- Grommet version:
- Browser Name and version:
- Operating System and version (desktop or mobile):

**Comment by eatyourgreens** [comment]
Outside of a screenreader, it's not obvious to a keyboard user that you need to switch from tab to cursor keys in order to navigate the individual bars. Tabbing through the page, it's easy to conclude that the chart isn't accessible from the keyboard.

Making each bar focusable, with an accessible name and role for each bar, would solve that and also make the bars accessible in VoiceOver/NVDA/JAWS etc.

**Comment by eatyourgreens** [comment]
Examining a Grommet bar chart on the Zooniverse site, I see that it uses a labelled div as the chart container. The ARIA spec explicitly forbids labels on generic roles, so this will be ignored by most AT. I'm not sure if this is a bug in the Grommet code, or something added by Zooniverse developers. cc. @mcbouslog 

<img width="502" alt="The Firefox DOM inspector open to show a div with an aria-label added. This doesn't make the div accessible." src="https://github.com/user-attachments/assets/1c4a7a3e-89c9-4b5f-b280-36e399c0646c">

**Comment by jcfilben** [comment]
Hey @eatyourgreens, thanks for reporting it looks like we have a few additional accessibility bugs on the DataChart component.

In regards to your comment about tab vs arrow keys I think using arrow keys to navigate through the individual chart points/bars is nice because it doesn't clutter the tab-index and makes it easy for the keyboard user to move past the DataChart without tabbing to every item in the chart. A good resource for when to use tab vs arrows is https://www.w3.org/WAI/ARIA/apg/patterns/. I do think we can make it more clear to the user how to navigate. DataChart is an older component, I think there are lot of accessibility improvements we could make.

**Comment by eatyourgreens** [comment]
@jcfilben Thanks. The ARIA design patterns are a great resource. I’ve used them previously to add the combobox pattern in #6079. The data chart isn’t working in VoiceOver because it doesn’t follow an accessible pattern. Making the detail bars focusable, with accessible roles and names, should fix that. 

Cursor key navigation would be appropriate for a menu or select, where there’s an established convention of using the cursor keys to navigate a list. Even then, you still need to markup the data chart as a grouped control, with the appropriate roles and states, and pass focus to the active bar when you press left or right. A screen reader user, or a keyboard user in general, will use the most appropriate keys for a control. It’s not clear to me that they would understand that they need to switch from tab to cursor keys here. 

There’s an analogous situation in usability testing of tabbed interfaces. Tabs should be controlled by the cursor keys, but many end users expect to be able to tab between tabs, and become frustrated when that doesn’t work. In fact, I think the Grommet Tabs component might ignore the cursor keys (it’s been a while since I’ve used it.)

For bar charts, I think the least confusing UX would be to allow both tab and cursor key navigation. If it’s intended to be navigated via the cursor keys in a screen reader, then both chart and bars must be announced as roles that blind and visually impaired users associate with the cursor keys.
